SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -565,6 +565,16 @@ int64 USpatialActorChannel::ReplicateActor()
 		}
 	}
 
+	if (Actor->GetIsHiddenDirty())
+	{
+		if (!Actor->GetRootComponent() || !Actor->GetRootComponent()->IsCollisionEnabled())
```

Comment:
Hmmmmmmmmmmmmmmmmmmm. So in native `AActor::IsNetRelevantFor` will return `false` `if(IsHidden() && ((!RootComponent || !RootComponent->IsCollisionEnabled()))` right. So that means when `IsHidden` is set we don't want clients to get replication updates when `(!RootComponent || !RootComponent->IsCollisionEnabled())`. But if we do have `bHidden` set and either `(RootComponent || RootComponent->IsCollisionEnabled())` then the server will default to normal distance checking for whether to replicate an actor to a client. 

To achieve the same effect I think we need to modify this logic such that we set the hidden component on the entity when `IsHidden() && ((!RootComponent || !RootComponent->IsCollisionEnabled())` but remove it when `IsHidden() && ((RootComponent || RootComponent->IsCollisionEnabled())` or `!IsHidden()`. Since this function handles updating the entities hidden component in all cases it currently doesn't achieve that.

Reply:
To simplify. In Native the server can control whether to replicate a hidden entity or not each tick based on the hidden flag and the root component. In Spatial we control whether a client sees updates about a hidden entity based on the hidden component. Since we can't check every tick we have to make sure the hidden component only exists on the entity in the exact same cases as when native would NOT replicate (`!IsNetRelevantFor`) an actor to a client

Reply:
Sorry by hidden component on the entity I meant the visible component on the entity. Forgot it was switched for our side :D

Reply:
oh yeah, you are right! that was working before that additional logic about the Collision checks :/ Btw if we want to do exactly what native unreal does do we also want to check if the actor is a PlayerController or a Pawn? As it seems they implement their own IsNetRelevantFor implementation.